src/gather_classes.rs

```diff
@@ -0,0 +1,732 @@
+// Contains functions for the gather-class-hashes sub command.
+
+// todo: Make async.
+
+use crate::{
+    io::{
+        self, deserialize_or, try_deserialize, try_deserialize_or, try_get_block_num_from_path,
+        try_serialize,
+    },
+    juno_manager::ManagerError,
+    trace_comparison::EMPTY,
+    utils::{self, val_or_err},
+};
+
+use glob::glob;
+use log::{debug, error, info, warn};
+use serde::{de::Visitor, Deserialize, Serialize};
+use serde_with::serde_as;
+
+use crate::trace_comparison::{DIFFERENT, SAME};
+
+use std::{
+    cmp::Ordering,
+    collections::{HashMap, HashSet},
+    io::{Read, Write},
+    path::{Display, Path, PathBuf},
+    str::FromStr,
+};
+
+use anyhow::{anyhow, Context};
+use itertools::Itertools;
+use starknet::{
+    contract,
+    core::types::{FieldElement, TransactionTraceWithHash},
+    macros::felt_hex,
+};
+use std::fs::OpenOptions;
+
+use crate::{
+    graph::{self, DependencyMap},
+    transaction_simulator::BlockSimulationReport,
+};
+
+use serde_json::{
+    json,
+    map::{Entry, Iter},
+    Value,
+};
+
+type ContractHash = FieldElement;
+type Selector = FieldElement;
+
+#[derive(Hash, Eq, PartialEq, Ord, PartialOrd, Clone)]
+struct EntryPoint {
+    contract_hash: ContractHash,
+    selector: Selector,
+}
+impl EntryPoint {
+    fn to_string(&self) -> String {
+        format!(
+            "({}, {})",
+            self.contract_hash.to_string(),
+            self.selector.to_string()
+        )
+    }
+
+    fn entry_point_from_str(entry_point_str: &str) -> Result<Self, anyhow::Error> {
+        let re = regex::Regex::new(r"\((?<contract_hash>.*?), (?<selector>.*?)\)")?;
+        match re.captures(entry_point_str) {
+            None => Err(anyhow!(
+                "Unexpected format for string: `{}`",
+                entry_point_str
+            )),
+            Some(caps) => Ok(Self {
+                contract_hash: FieldElement::from_str(&caps["contract_hash"])?,
+                selector: FieldElement::from_str(&caps["selector"])?,
+            }),
+        }
+    }
+}
```

Comment:
I don't understand why are you converting things to string and back

Reply:
Removed after changing EntryPoint to just a selector.